overlord/snapstate/backend/snapdata.go

```diff
@@ -116,10 +116,16 @@ func removeDirs(dirs []string) error {
 // snapDataDirs returns the list of data directories for the given snap version
 func snapDataDirs(snap *snap.Info, opts *dirs.SnapDirOptions) ([]string, error) {
 	// collect the directories, homes first
-	found, err := filepath.Glob(snap.DataHomeDir(opts))
-	if err != nil {
-		return nil, err
+	var found []string
+
+	for _, entry := range snap.DataHomeDirs(opts) {
+		entryPaths, err := filepath.Glob(entry)
```

Comment:
this part doesn't seem to be covered, could you please add/update unit tests to exercise this part? check `TestRemoveSnapData` for reference.